### PR TITLE
Return of the facehuggers

### PR DESCRIPTION
**Pull request by Alberyk**
This pr re-adds facehuggers, taken away by the cold hands of the baymerge, and the alien infection, besides fixing some issues with it. Also, removed gibbing when the chestbursters burts out of the host and turns the alien embryo into a proper organ.

### CODE REVIEW

File: code/game/objects/structures/alien/egg.dm [hunk 1]
```diff
@@ -16,82 +23,73 @@
...
+	return
+
+/obj/structure/alien/egg/proc/Burst(var/kill = 1) //drops and kills the hugger if any is remaining
+	if(status == GROWN || status == GROWING)
+		var/obj/item/clothing/mask/facehugger/child = GetFacehugger()
+		icon_state = "egg_hatched"
+		flick("egg_opening", src)
+		status = BURSTING
+
+		spawn(15)
```
Comment (Lohikar) [inline code/game/objects/structures/alien/egg.dm]: _hiss_

File: code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm [hunk 2]
```diff
@@ -0,0 +1,154 @@
+/obj/item/alien_embryo
+	name = "alien embryo"
+	desc = "All slimy and yuck."
+	icon = 'icons/mob/alien.dmi'
+	icon_state = "larva0_dead"
+	var/mob/living/affected_mob
+	var/stage = 0
+	var/infected = FALSE
+
+/obj/item/alien_embryo/New()
```
Comment (Lohikar) [inline code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm]: _hiss_

File: code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+		START_PROCESSING(SSprocessing, src)
+		AddInfectionImages(affected_mob)
+	else
+		qdel(src)
+
+/obj/item/alien_embryo/Destroy()
+	if(affected_mob)
+		affected_mob.status_flags &= ~(XENO_HOST)
+		RemoveInfectionImages(affected_mob)
+	..()
```
Comment (Lohikar) [inline code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm]: Return parent.

File: code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	if(loc != affected_mob)
+		affected_mob.status_flags &= ~(XENO_HOST)
+		STOP_PROCESSING(SSprocessing, src)
+		RemoveInfectionImages(affected_mob)
+		affected_mob = null
+		return
+
+	if(stage < 5 && prob(3))
+		stage++
+		spawn(0)
```
Comment (Lohikar) [inline code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm]: _hiss_

File: code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+		picked = affected_mob.key
+	else
+		stage = 4 // Let's try again later.
+		return
+
+	if(affected_mob.lying)
+		affected_mob.overlays += image('icons/mob/alien.dmi', loc = affected_mob, icon_state = "burst_lie")
+	else
+		affected_mob.overlays += image('icons/mob/alien.dmi', loc = affected_mob, icon_state = "burst_stand")
+	spawn(6)
```
Comment (Lohikar) [inline code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm]: _hiss_

File: code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+	else if(affected_mob.client)
+		picked = affected_mob.key
+	else
+		stage = 4 // Let's try again later.
+		return
+
+	if(affected_mob.lying)
+		affected_mob.overlays += image('icons/mob/alien.dmi', loc = affected_mob, icon_state = "burst_lie")
+	else
+		affected_mob.overlays += image('icons/mob/alien.dmi', loc = affected_mob, icon_state = "burst_stand")
```
Comment (Lohikar) [inline code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm]: `add_overlay()`, preferably using priority.

File: code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+	if(candidates.len)
+		picked = pick(candidates)
+	else if(affected_mob.client)
+		picked = affected_mob.key
+	else
+		stage = 4 // Let's try again later.
+		return
+
+	if(affected_mob.lying)
+		affected_mob.overlays += image('icons/mob/alien.dmi', loc = affected_mob, icon_state = "burst_lie")
```
Comment (Lohikar) [inline code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm]: and here

File: code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+/obj/item/alien_embryo/proc/RefreshInfectionImage()
+
+	for(var/mob/living/carbon/alien in player_list)
+
+		if(!locate(/obj/item/organ/xenos/hivenode) in alien.internal_organs)
+			continue
+
+		if(alien.client)
+			for(var/image/I in alien.client.images)
+				if(dd_hasprefix_case(I.icon_state, "infected"))
```
Comment (Lohikar) [inline code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm]: Yeah, no. Find a better way than qdeling images, it's laggy and doesn't work.

Comment (Alberyk) [reply code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm]: Not really sure how to handle the removal of those images right now.

**Comment by Alberyk** [comment]
Replaced the qdel of images and fixed some other issues, should be good for another review.

File: code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm (hunk 8 above)
Comment (Alberyk) [reply code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm]: Should be fixed now.

**Pull request merged**